File: 0080 Remove Duplicates from Sorted Array II.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length <= 2) return nums.length;
        int last = nums[0];
        int cnt = 1;
        int j = 1;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i] != last) {
                nums[j++] = nums[i];
                last = nums[i];
                cnt = 1;
            } else if (cnt < 2) {
                nums[j++] = nums[i];
                ++cnt;
            }
        }
        return j;
    }
}

/* Solution from StefanPochmann
public int removeDuplicates(int[] nums) {
    int i = 0;
    for (int n : nums)
        if (i < 2 || n > nums[i-2])
            nums[i++] = n;
    return i;
}
*/

File: 0080 Remove Duplicates from Sorted Array II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) < 2: return len(nums)
        last = nums[0]
        cnt, j = 1, 1
        for i in range(1, len(nums)):
            if nums[i] != last:
                nums[j], last = nums[i], nums[i]
                cnt = 1
                j += 1
            elif cnt < 2:
                nums[j] = nums[i]
                j += 1
                cnt += 1
        return j

''' Solution from StefanPochmann
def removeDuplicates(self, nums):
    i = 0
    for n in nums:
        if i < 2 or n > nums[i-2]:
            nums[i] = n
            i += 1
    return i
'''